### PR TITLE
sink/mq(ticdc): make the topic manager use the correct number of partitions

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/5076

### What is changed and how it works?

make the topic manager use the correct number of partitions.

It uses the wrong number of partitions when kafka turns on auto-create configuration.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change

Side effects

None

Related changes

None

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by zhaoxinyu — APPROVED** [review]

File: cdc/sink/manager/kafka/manager.go [hunk 1]
```diff
@@ -106,27 +106,27 @@ func (m *TopicManager) tryUpdatePartitionsAndLogging(topic string, partitions in
...
 		log.Info(
 			"store topic partition number",
-			zap.String("duration", topic),
+			zap.String("topic", topic),
 			zap.Int32("partitionNumber", partitions),
 		)
 	}
 }
 
 // CreateTopic creates a topic with the given name.
```
Comment (asddongmen) [inline cdc/sink/manager/kafka/manager.go:115]: Can you add a comment to explain the meaning of the return `int32` value?

Comment (hi-rustin) [reply cdc/sink/manager/kafka/manager.go]: Good catch! Done!

**Review by asddongmen — APPROVED** [review]

**Pull request merged**